lib/mongo/scrollable.rb

```diff
@@ -16,21 +16,37 @@ def scroll(cursor_or_type = nil, options = nil, &_block)
       cursor_options = { field_name: scroll_field, direction: scroll_direction }.merge(options)
       cursor = cursor && cursor.is_a?(cursor_type) ? cursor : cursor_type.new(cursor, cursor_options)
       raise_mismatched_sort_fields_error!(cursor, cursor_options) if different_sort_fields?(cursor, cursor_options)
-      # make a view
-      view = Mongo::Collection::View.new(
-        view.collection,
-        view.selector.merge(cursor.criteria),
-        sort: (view.sort || {}).merge(_id: scroll_direction),
-        skip: skip,
-        limit: limit
-      )
+
+      records = nil
+      if cursor.previous && limit
+        # scroll backards by reversing the sort order, limit and then reverse again
+        pipeline = [
+          { '$match' => view.selector.merge(cursor.criteria) },
+          { '$sort' => { scroll_field => -scroll_direction } },
+          { '$limit' => limit },
+          { '$sort' => { scroll_field => scroll_direction } }
+        ]
+        aggregation_options = view.options.except(:sort)
+        records = view.aggregate(pipeline, aggregation_options)
+      else
+        # make a view
+        records = Mongo::Collection::View.new(
+          view.collection,
+          view.selector.merge(cursor.criteria),
+          sort: (view.sort || {}).merge(_id: scroll_direction),
+          skip: skip,
+          limit: limit
+        )
+      end
```

Comment:
This : 

1. Select records before the limit in the criteria
2. Reverse the sort
3. Take only the number of records in `limit`
4. Resort the order

Given that, if we have this list : 

```
1
2
3
4
5 # <= Cursor here
6 
```

We want to have 2 records before the cursor and keep the order. Returned values have to be `3, 4`. 


To do that, we first select only records be the cursor :

```
1
2
3
4
```

We reverse the selection :  

```
4
3
2
1
```

Take only the two first values : 

```
4
3
```

And finally revert the sorting : 

```
3
4
```

To avoid using aggregation, another solution was possible by sorting first with mongo and the reverse the records to display in the correct order
Another possible solution was to do the same in pure Ruby : 

```
view.sort[scroll_field] = -scroll_direction if cursor.previous
records = Mongo::Collection::View.new(
  view.collection,
  view.selector.merge(cursor.criteria),
  sort: (view.sort || {}).merge(_id: -scroll_direction),
  skip: skip,
  limit: limit
)
records.to_a.sort { |h1, h2| h1[scroll_field] <=> h2[scroll_field] }.each do |record|
  # ...
end
```

Even if this version is simpler, I chose to use aggregates for performance reason.